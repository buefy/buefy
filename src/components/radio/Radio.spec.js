import { shallowMount } from '@vue/test-utils'
import BRadio from '@components/radio/Radio'

let wrapper

describe('BRadio', () => {
    beforeEach(() => {
        wrapper = shallowMount(BRadio)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('BRadio')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has an input radio', () => {
        expect(wrapper.contains('label input[type=radio]')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        wrapper.setProps({ value: true })
        expect(wrapper.vm.computedValue).toBeTruthy()
        wrapper.vm.computedValue = false
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(false)
    })

    it('method focus() gives focus to the input element', (done) => {
        wrapper.vm.$refs.input.focus = jest.fn()
        wrapper.vm.focus()
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled()
            done()
        })
    })
})
