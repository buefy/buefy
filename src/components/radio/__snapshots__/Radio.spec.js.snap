// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BRadio render correctly 1`] = `<label class="b-radio radio"><input type="radio" value="false"> <span class="check"></span> <span class="control-label"></span></label>`;
